src/components/Card.js

```diff
@@ -1,48 +1,55 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
+  constructor(
+    { name, link, _id },
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick
+  ) {
     this._name = name;
     this._link = link;
+    this._id = _id;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
 
-    this._cardElement = this._getTemplate();
-    this._cardImageEl = this._cardElement.querySelector(".card__image");
-    this._cardTitleEl = this._cardElement.querySelector(".card__title");
-    this._trashBtnEl = this._cardElement.querySelector(".card__delete-button");
-    this._likeBtnEl = this._cardElement.querySelector(".card__like-button");
+    this._element = this._getTemplate();
+    this._image = this._element.querySelector(".card__image");
+    this._title = this._element.querySelector(".card__title");
+    this._deleteButton = this._element.querySelector(".card__delete-button");
+    this._likeButton = this._element.querySelector(".card__like-button");
   }
 
   getView() {
-    this._cardImageEl.src = this._link;
-    this._cardImageEl.alt = this._name;
-    this._cardTitleEl.textContent = this._name;
-
+    this._image.src = this._link;
+    this._image.alt = this._name;
+    this._title.textContent = this._name;
     this._setEventListeners();
-    return this._cardElement;
+    return this._element;
   }
 
   _getTemplate() {
     const template = document
       .querySelector(this._cardSelector)
-      .content.querySelector(".card")
-      .cloneNode(true);
-    return template;
+      .content.querySelector(".card");
+    return template.cloneNode(true);
   }
 
   _setEventListeners() {
-    this._likeBtnEl.addEventListener("click", () => this._toggleLike());
-    this._trashBtnEl.addEventListener("click", () => this._deleteCard());
-    this._cardImageEl.addEventListener("click", () => {
+    this._likeButton.addEventListener("click", () => this._toggleLike());
```

Comment:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.